.github/workflows/ci.yml

```diff
@@ -1,9 +1,16 @@
 name: CI
 
 on:
+  push:
+    branches:
+      - main
   pull_request:
     types: [opened, synchronize, reopened]
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
```

Comment:
This is [Only cancel in-progress jobs or runs for the current workflow](https://docs.github.com/en/enterprise-cloud@latest/actions/using-jobs/using-concurrency#example-only-cancel-in-progress-jobs-or-runs-for-the-current-workflow)

I used these [GitHub Context](https://docs.github.com/ko/actions/learn-github-actions/contexts#github-context)
- github.ref
- github.workflow